src/NossoMercadoLivreAPI.Application/Controllers/UserController.cs

```diff
@@ -1,106 +1,65 @@
 ﻿using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Mvc;
 using System;
-using System.Collections.Generic;
-using NossoMercadoLivreAPI.Application.Controllers.Base;
-using NossoMercadoLivreAPI.Domain.Interfaces.Services;
-using NossoMercadoLivreAPI.Domain.Request;
-using NossoMercadoLivreAPI.Domain.Response;
 using System.Threading.Tasks;
+using NossoMercadoLivreAPI.Domain.Interfaces.Repositories;
+using NossoMercadoLivreAPI.Domain.Entities;
+using NossoMercadoLivreAPI.Domain.Request;
 using FluentValidation;
+using FluentValidation.Results;
+using NossoMercadoLivreAPI.Util;
 
 namespace NossoMercadoLivreAPI.Application.Controllers
 {
+    [Produces("application/json")]
     [Route("api/[controller]")]
-    public class UserController : BaseController
+    [ApiController]
+    public class UserController : ControllerBase
     {
-        private readonly IUserService _userService;
-        
-        public UserController(IUserService userService)
+        private readonly IValidator<UserRequest> _validator;
+
+        public UserController(IValidator<UserRequest> validator)
```

Comment:
*Usamos tudo que conhecemos que está pronto. Só fazemos código do zero se for estritamente necessário.*
+
*Avaliamos de forma lógica o nível de complicação de cada trecho de código. Aqui não tem espaço para feeling. Usamos a teoria da carga cognitiva e medimos o nível de pontos de entendimentos necessário por trecho.*

Pq temos uma interface com uma implementação se a validação através de atributos/annotations/decorators na ViewModel já funciona por padrão? Você anota lá e usa o ModelState para verificar.  Isso aumenta a dificuldade de entendimento do sistema e não parece trazer nenhuma vantagem para compensar :).